av/container/core.pyx

```diff
@@ -167,6 +206,11 @@ cdef class Container(object):
         self.ptr.flags |= lib.AVFMT_FLAG_GENPTS
         self.ptr.max_analyze_duration = 10000000
 
+        if audio_codec is None:
```

Comment:
Input validation needs to be done much earlier. If you encounter a KeyError here, you're going to leak memory as we have already started allocating structures. I'm not even sure it makes sense to handle the None case, as I believe `AV_CODEC_ID_NONE` is already the default?